File: VB/MinimalisticReportPreviewDemo/MyDocumentPreviewControl.vb
Imports DevExpress.Xpf.Printing
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.Xpf.Printing.PreviewControl
Imports DevExpress.XtraPrinting

Namespace DXSample
	Public Class MyDocumentPreviewControl
		Inherits DocumentPreviewControl

		Public Overrides Sub Print()
			Dim pt As New PrintTool(Document.PrintingSystem)
			AddHandler pt.PrintingSystem.StartPrint, AddressOf PrintingSystem_StartPrint
			AddHandler pt.PrintingSystem.PrintProgress, AddressOf PrintingSystem_PrintProgress
			pt.PrintDialog()
		End Sub
		Public Overrides Sub PrintDirect(Optional ByVal printerName As String = Nothing)
			Dim pt As New PrintTool(Document.PrintingSystem)
			AddHandler pt.PrintingSystem.StartPrint, AddressOf PrintingSystem_StartPrint
			AddHandler pt.PrintingSystem.PrintProgress, AddressOf PrintingSystem_PrintProgress
			pt.Print()
		End Sub
		Private Sub PrintingSystem_PrintProgress(ByVal sender As Object, ByVal e As PrintProgressEventArgs)

		End Sub
		Private Sub PrintingSystem_StartPrint(ByVal sender As Object, ByVal e As PrintDocumentEventArgs)

		End Sub
	End Class
End Namespace

File: CS/MinimalisticReportPreviewDemo/MyDocumentPreviewControl.cs
using DevExpress.Xpf.Printing;
using DevExpress.XtraPrinting;

namespace DXSample {
    public class MyDocumentPreviewControl : DocumentPreviewControl {
        public override void Print() {
            PrintTool pt = new PrintTool(Document.PrintingSystem);
            pt.PrintingSystem.StartPrint += new PrintDocumentEventHandler(PrintingSystem_StartPrint);
            pt.PrintingSystem.PrintProgress += new PrintProgressEventHandler(PrintingSystem_PrintProgress);
            pt.PrintDialog();
        }
        public override void PrintDirect(string printerName = null) {
            PrintTool pt = new PrintTool(Document.PrintingSystem);
            pt.PrintingSystem.StartPrint += new PrintDocumentEventHandler(PrintingSystem_StartPrint);
            pt.PrintingSystem.PrintProgress += new PrintProgressEventHandler(PrintingSystem_PrintProgress);
            pt.Print();
        }
        void PrintingSystem_PrintProgress(object sender, PrintProgressEventArgs e) {

        }
        void PrintingSystem_StartPrint(object sender, PrintDocumentEventArgs e) {
        
        }
    }
}
